### DISCUSSION TITLE
How to generate a default template of a control?

### DISCUSSION THREAD

**Discussion by msnoshain** [AvaloniaUI/Avalonia#8740] [Q&A] [answered]
It will be convenient to rewrite some controls if Avalonia can generate default templates for user to edit. (just like in WPF)

**Comment by maxkatz6** [comment, marked as answer]
Avalonia is open source, unlikely WPF (in the past), so you can find all templates in the source code. Per specific version and with way more historical information.
For instance https://github.com/AvaloniaUI/Avalonia/tree/release/0.10.17/src/Avalonia.Themes.Fluent/Controls

**Reply by msnoshain** [reply]
This seems to be the best way at present 👍
